### PR TITLE
At ionosphere, retain only boundary-normal components of copied perB values

### PR DESCRIPTION
**Pull request by markusbattarbee**
Testing is underway but looks promising in reverting near-inner boundary behaviour to Vlasiator 4 -style. Perhaps code approach can still be cleaned up?

### CODE REVIEW

File: fieldsolver/ldz_magnetic_field.cpp [hunk 1]
```diff
@@ -192,6 +192,36 @@ void propagateSysBoundaryMagneticField(
...
+ * \param perBGrid fsGrid holding the perturbed B quantities at runge-kutta t=0
+ * \param perBDt2Grid fsGrid holding the perturbed B quantities at runge-kutta t=0.5
+ * \param technicalGrid fsGrid holding technical information (such as boundary types)
+ * \param i,j,k fsGrid cell coordinates for the current cell
+ * \param sysBoundaries System boundary conditions existing
+ * \param RKCase Element in the enum defining the Runge-Kutta method steps
+ *
+ * \sa propagateMagneticFieldSimple propagateMagneticField
+ */
+void projectSysBoundaryMagneticField(
```
Comment (ursg) [inline fieldsolver/ldz_magnetic_field.cpp:208]: One thing that really confused me when looking at this diff for the first time is the two different meanings of the word "project" in the source code.

There is Project::setProjectBField, which deals with the magnetic field set by a project.
And now there is projectSysBoundaryMagneticField, which dieals with the magnetic field *projection*.

Should we, to minimize confusion, maybe try to consistently use the form "projection" here? So this identifier would be sysBoundaryMagneticFieldProjection?

File: sysboundary/ionosphere.cpp [hunk 2]
```diff
@@ -707,6 +705,41 @@ namespace SBC {
       }
    }
 
+   /*! We want here to
+    *
+    * -- Retain only the boundary-normal projection of perturbed face B
+    */
+   void Ionosphere::fieldSolverBoundaryCondMagneticFieldProject(
```
Comment (ursg) [inline sysboundary/ionosphere.cpp:712]: Likewise here, I suggest Project → Projection

**Review by ursg — APPROVED** [review]
Functionally approved!
(But note my comments about the terminology of "Project" vs "Projection")

**Comment by markusbattarbee** [comment]
Fair enough, changed! I'll still recompile and run a testpackage, though of course now we can expect diffs whenever there's an ionosphere involved :)

**Comment by markusbattarbee** [comment]
Confirmed, it works. And diffs only in magnetospheric runs.

**Pull request merged**